src/AbstractApplication.php

```diff
@@ -105,20 +106,27 @@
      * @var Collection
      */
     protected $packages;
-    
+
     /**
      * AbstractApplication constructor.
      *
      * @param ClassLoader|null $autoloader
+     * @param EventsHandler|null $eventsHandler
+     * @throws \ObjectivePHP\Events\Exception
+     * @throws \ObjectivePHP\Primitives\Exception
+     * @throws \ObjectivePHP\ServicesFactory\Exception\ServiceNotFoundException
      */
-    public function __construct(ClassLoader $autoloader = null)
+    public function __construct(ClassLoader $autoloader = null, EventsHandler $eventsHandler = null)
```

Comment:
EventsHandlerInterface ?

In fact in think your right here: it's difficult to change the event handler after the Application construction as BOOTSTRAP_INIT and BOOTSTRAP_DONE will already triggered.

@gauthier Could we move this trigger outside the constructor ?

Reply:
I did this because the getEventshandler method create a new instance (that I don't want) in the constructor.

Reply:
If it's not moved outside the constructor, the setter method setEventHandler is useless.